tokio/src/io/async_fd.rs

```diff
@@ -85,14 +86,28 @@ impl<T: AsRawFd> AsyncFd<T> {
     where
         T: AsRawFd,
     {
-        Self::new_with_handle(inner, Handle::current())
+        Self::with_interest(inner, ALL_INTEREST)
+    }
+
+    #[inline]
+    /// Creates new instance as `new` with additional ability to customize interest,
+    /// allowing to specify whether file descriptor will be polled for read, write or both.
+    pub fn with_interest(inner: T, interest: Interest) -> io::Result<Self>
+    where
+        T: AsRawFd,
+    {
+        Self::new_with_handle_and_interest(inner, Handle::current(), interest)
     }
 
-    pub(crate) fn new_with_handle(inner: T, handle: Handle) -> io::Result<Self> {
+    pub(crate) fn new_with_handle_and_interest(
```

Comment:
We may also want to rename this to `with_handle_and_interest` for consistency, but definitely not a blocker. I will merge this shortly if nobody has any other comments.